src/arena.rs

```diff
@@ -139,6 +139,11 @@ impl<T> Arena<T> {
             self.append(value)
         }
     }
+
+    /// Get a mutable reference to an element in the arena.
+    pub fn mutate(&mut self, handle: Handle<T>) -> &mut T {
```

Comment:
The function name is a bit misleading, as I initially thought this would mutate the handle directly, while the intend is to get a mutable reference to it and then mutate a value in the handle.

I am not sure what we would want to call it, maybe `get_mut_handle`?